single-node-refactor/src/Solvers/SGH_solver/src/sgh_solve.cpp

```diff
@@ -232,6 +232,17 @@ void SGH::execute(simulation_parameters_t& sim_param, mesh_t& mesh, node_t& node
                           rk_alpha);
             }
 
+            // ---- apply contact boundary conditions to the boundary patches----
+            if (doing_contact)  // Structuring it like this to avoid having to sort() everytime.
+            {
+                boundary_contact(mesh, node, corner, dt);
+                // Todo: Ask about why the coordinates are updating every other step?
```

Comment:
Which coordinates?   If that is happening, you may have found a bug. :)

Reply:
I was going to get to this in the future, but the node.coords(0, *, *) and node.vel(0, *, *) are only changing every other step. You can verify this by uncommenting those lines beneath. I was not sure if this is the expected behavior, but I think there's a step missing where after it solves, it should shift the points from node.coords(1, *, *) to node.coords(0, *, *). I think it's only doing that shift every other step. I saw a `% 20` branch somewhere in there, so there is some kind of logic with doing things differently for an even number of steps/cycles.

Reply:
Ahh. I understand.  Ill explain this at the meeting. It has to do with our time integration scheme.